File: LivrusVirtus/wxWindows/wxRuby/Splash.rb
require 'wxruby'
require 'Login'

$fSplash = nil

class FrmSplash < Wx::Dialog
  ID_TIMER = 1000
  def initialize()
    super(nil, -1, "Livrus Virtus", Wx::Point.new(0, 0), 
      Wx::Size.new(255, 73), Wx::THICK_FRAME, "") 
    set_client_size(Wx::Size.new(255, 73))
    centre()
    
    @labBmp = Wx::StaticBitmap.new(self, -1, 
      Wx::Bitmap.new("xpms\\livrusvirtus.xpm", Wx::BITMAP_TYPE_ANY),
      Wx::Point.new(0, 0), Wx::Size.new(254, 72))

    @mTimer = Wx::Timer.new(self, ID_TIMER)
    @mTimer.start(1000)

    evt_close {|event| on_close(event)}
    evt_timer(ID_TIMER) {onTimer}
  end

  def onTimer
    @mTimer.stop()
    self.close()
  end

  def on_close(event)
    fLogin = FrmLogin.new
    fLogin.show(true)
    self.destroy()
    event.skip()
  end

end

File: LivrusVirtus/wxWindows/wxLua/Splash.lua
ID_TIMER = 100
ID_FRMSPLASH = 101

frmSplash_mTimer = nil

function frmSplash_Criar()
  local fSplash = wx.wxDialog(wx.wxNull, ID_FRMSPLASH, STR_TITULO, wx.wxPoint(0, 0), 
      wx.wxSize(255, 73), wx.wxNO_BORDER, "")   

  fSplash:SetClientSize(wx.wxSize(255, 73))
  fSplash:Centre()

  bmp_splash = wx.wxBitmapFromXPMData(livrusvirtus_xpm)
  frmSplash_labBmp = wx.wxStaticBitmap(fSplash, -1, bmp_splash,
    wx.wxPoint(0, 0), wx.wxSize(254, 72))

  frmSplash_mTimer = wx.wxTimer(fSplash, ID_TIMER)
  frmSplash_mTimer:Start(1000)

  fSplash:ConnectEvent(ID_TIMER, wx.wxEVT_TIMER, 
    function (event)
      frmSplash_mTimer:Stop()
      fSplash:Close()     
    end )

  fSplash:ConnectEvent(ID_FRMSPLASH,wx.wxEVT_CLOSE_WINDOW,
    function (event)
      frmLogin:Show(wx.TRUE)
      --fSplash:Destroy()
      event:Skip()
    end )
  
  return fSplash
end

frmSplash = frmSplash_Criar()
